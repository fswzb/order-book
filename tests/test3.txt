{"type": "Limit", "order": {"direction": "Sell", "id": 1, "price": 101, "quantity": 20000}}
{"type": "Limit", "order": {"direction": "Buy", "id": 2, "price": 99, "quantity": 50000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 3, "price": 100, "quantity": 10000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 4, "price": 100, "quantity": 7500}}
{"type": "Limit", "order": {"direction": "Buy", "id": 5, "price": 98, "quantity": 25500}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 6, "price": 100, "quantity": 100000, "peak": 10000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 7, "price": 100, "quantity": 10000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 8, "price": 100, "quantity": 11000}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 9, "price": 100, "quantity": 50000, "peak": 20000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 10, "price": 100, "quantity": 35000}}
{"type": "Limit", "order": {"direction": "Sell", "id": 11, "price": 100, "quantity": 20000}}
