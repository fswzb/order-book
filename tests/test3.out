{"buyOrders": [], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrders": [{"id": 2, "price": 99, "quantity": 50000}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrders": [{"id": 2, "price": 99, "quantity": 50000}], "sellOrders": [{"id": 3, "price": 100, "quantity": 10000}, {"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrders": [{"id": 2, "price": 99, "quantity": 50000}], "sellOrders": [{"id": 3, "price": 100, "quantity": 10000}, {"id": 4, "price": 100, "quantity": 7500}, {"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrders": [{"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 3, "price": 100, "quantity": 10000}, {"id": 4, "price": 100, "quantity": 7500}, {"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrderId": 6, "sellOrderId": 3, "price": 100, "quantity": 10000}
{"buyOrderId": 6, "sellOrderId": 4, "price": 100, "quantity": 7500}
{"buyOrders": [{"id": 6, "price": 100, "quantity": 10000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrderId": 6, "sellOrderId": 7, "price": 100, "quantity": 10000}
{"buyOrders": [{"id": 6, "price": 100, "quantity": 10000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrderId": 6, "sellOrderId": 8, "price": 100, "quantity": 10000}
{"buyOrderId": 6, "sellOrderId": 8, "price": 100, "quantity": 1000}
{"buyOrders": [{"id": 6, "price": 100, "quantity": 9000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrders": [{"id": 6, "price": 100, "quantity": 9000}, {"id": 9, "price": 100, "quantity": 20000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrderId": 6, "sellOrderId": 10, "price": 100, "quantity": 9000}
{"buyOrderId": 9, "sellOrderId": 10, "price": 100, "quantity": 20000}
{"buyOrderId": 6, "sellOrderId": 10, "price": 100, "quantity": 6000}
{"buyOrders": [{"id": 6, "price": 100, "quantity": 4000}, {"id": 9, "price": 100, "quantity": 20000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
{"buyOrderId": 6, "sellOrderId": 11, "price": 100, "quantity": 4000}
{"buyOrderId": 9, "sellOrderId": 11, "price": 100, "quantity": 16000}
{"buyOrders": [{"id": 9, "price": 100, "quantity": 4000}, {"id": 6, "price": 100, "quantity": 10000}, {"id": 2, "price": 99, "quantity": 50000}, {"id": 5, "price": 98, "quantity": 25500}], "sellOrders": [{"id": 1, "price": 101, "quantity": 20000}]}
